src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,53 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNullUser(user);
+        checkFieldsForNullValue(user);
+        checkFieldsForMinValues(user);
+        checkUserExistInStorage(user);
```

Comment:
I think, you should move it after checkNullUser. Why do you need to check every field, if user already exists?

Reply:
I don't wanna "connect" to the storage if the user has non-valid fields. It's a redundant database call.